pkg/az/action.go

```diff
@@ -2,19 +2,21 @@ package az
 
 import (
 	"get.porter.sh/porter/pkg/exec/builder"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
 )
 
 var _ builder.ExecutableAction = Action{}
 var _ builder.BuildableAction = Action{}
 
 type Action struct {
 	Name  string
-	Steps []Step // using UnmarshalYAML so that we don't need a custom type per action
+	Steps []TypedStep // using UnmarshalYAML so that we don't need a custom type per action
```

Comment:
I'm going to play around with Go 1.18 and see if we can drastically simplify how we unmarshal mixin steps with generics.